File: 191 Number of 1 Bits/number_of_one_bits.cpp
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Number of 1 Bits
 *	https://leetcode.com/problems/number-of-1-bits/
 *	Runtime: 16 ms
 */
 
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count = 0;
        for (int i = 0; i < 32; i++) {
            count += n % 2;
            n /= 2;
        }
        return count;
    }
};

File: 191 Number of 1 Bits/number_of_one_bits.cs
/*	
 *	LeetCode Submissions by Xinyu Chen
 *	Number of 1 Bits
 *	https://leetcode.com/problems/number-of-1-bits/
 *	Runtime: 292 ms
 */
 
public class Solution {
    public int HammingWeight(uint n) {
        uint count = 0;
        while (n > 0)
        {
            count += n & 1;
            n >>= 1;
        }
        return (int)count;
    }
}
